plugins/kubernetes-backend/src/service/KubernetesProxy.test.ts

```diff
@@ -0,0 +1,328 @@
+/*
+ * Copyright 2022 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { getVoidLogger } from '@backstage/backend-common';
+import { ClusterDetails, KubernetesClustersSupplier } from '../types/types';
+import { KubernetesProxy, KubernetesProxyServices } from './KubernetesProxy';
+
+import { Request } from 'express';
+
+import 'buffer';
+
+jest.mock('node-fetch');
+const { Response } = jest.requireActual('node-fetch');
+
+import fetch from 'node-fetch';
+
+describe('KubernetesProxy', () => {
+  // @ts-ignore
+  let clientMock: any;
+  let sut: KubernetesProxy;
+
+  const buildEncodedRequest = (
+    clustersHeader: any,
+    query: string,
+    body?: any,
+  ): Request => {
+    const encodedQuery = encodeURIComponent(query);
+    const encodedClusters = Buffer.from(
+      JSON.stringify(clustersHeader),
+    ).toString('base64');
+
+    const req = {
```

Comment:
you can do `new Request(url, { headers: { ... } })` (same 1-or-2 argument shape as you give to `fetch`), would probably be best to use that constructor properly instead of faking it

Reply:
After fiddling with this, it's because the Express Request interface has no constructor, so creating a `new Request` instance falls back to the FetchAPI version, TS will error out in that case, so we can't use the `Request` constructor, `Request` from FetchAPI is different from `Request` from Express.

Reply:
Doubt it, Express request have some functions that the Request from FetchAPI doesn't, case in point the header function, in normal request is a map, but in Express is a getter function, which we fake with a switch case here